src/profile_manager/views.py

```diff
@@ -163,16 +163,28 @@ class ProfileDetail(NonPublicOnlyViewMixin, DetailView):
     @method_decorator(login_required)
     def dispatch(self, *args, **kwargs):
         # only allow the users to see their own profiles, or admins
-        profile_user = get_object_or_404(Profile, pk=self.kwargs.get('pk')).user
-        if profile_user == self.request.user or self.request.user.is_staff:
+        if self.request.user.is_staff:
+            return super().dispatch(*args, **kwargs)
+
+        profile = self.get_object()
+        if profile.user == self.request.user:
             return super().dispatch(*args, **kwargs)
 
         return redirect('quests:quests')
 
+    def get_object(self):
+        pk = self.kwargs.get('pk')
+        profile = self.request.user.profile
```

Comment:
Why is `profile = self.request.user.profile` added here if ProfileDetailOwn has it's own view?

Reply:
I was going through the code again and I kinda remember now that the sole reason I added this was just so profile doesn't get set to `None`.

I removed this and replaced it with raising an Http404 instead.